packages/terra-show-hide/src/ShowHide.jsx

```diff
@@ -55,9 +67,37 @@ const ShowHide = (props) => {
     preview,
     intl,
     isOpen,
+    focusRef,
     ...customProps
   } = props;
 
+  const theme = useContext(ThemeContext);
+  const contentRef = useRef(null);
+  const [containerIsActive, setContainerIsActive] = useState(false);
+
+  /**
+   * Upon showing hidden content, set activeElement to focusRef element if provided, othervise set it to content container.
+   */
+  useEffect(() => {
+    if (!isOpen) {
+      return;
+    }
+    if (focusRef?.current) {
+      focusRef?.current.focus();
+    } else if (contentRef?.current) {
+      setContainerIsActive(true);
+    }
+  }, [isOpen, focusRef, contentRef]);
+
+  useEffect(() => {
+    if (containerIsActive && isOpen && contentRef?.current) {
+      contentRef?.current?.focus();
+    }
+  }, [containerIsActive, isOpen]);
+
+  const id = uuidv4();
```

Comment:
I don't think you need to worry about pulling in this new library to generate a unique id.

Reply:
I thought that is what we were doing recently for uuid, but feel free to show me the example of how we are doing that lately. @sdadn - your input is also very appreciated as for what approach to choose.

Reply:
renamed to uniqueId in https://github.com/cerner/terra-core/pull/3838/commits/6ad93a1495c9721c4d1f7f0d046706d56c764413